bn-models/src/models/_lookups/ole-vessel.ts

```diff
@@ -0,0 +1,125 @@
+import { Base } from '../_base';
+import { BoatnetDate } from '../_common/index';
+
+export const DeclarationTypeName = 'ole-declaration';
+export const OLEVesselTypeName = 'ole-vessel';
+
+export interface Declaration extends Base {
+  declarationId?: number; // OLE declaration code (three digit for now)
+  observerStatus?: string; // type of observer coverage
+  efpStatus?: Boolean; // boolean, if declaration is under an EFP
+  activityDescrip?: string; // if 69 chosen, further description of declaration
+  confrimationNum?: number; // once declaration is in the system it is assigned a confirmation number
+  transactionDate?: BoatnetDate; // effective declaration start date
+  transactionType?: string; // tbd, but probably going to use it for keeping track of how the declaration was provided (app, phone, etc)
+  staffComments?: string;
+  userComments?: string;
+}
+
+export interface OLEVessel extends Base {  
```

Comment:
As a suggestion, since we already have a vessel.ts lookup which shares some of the same fields as OLEVessel: (vesselName, coastGuard#, stateReg#) you could create a base vessel interface with these fields then extend OLEVessel and vessel from that.

Reply:
Yeah, we briefly discussed this. I'm thinking it might be a future change we want to make, but not 100% sure on it right now.  There's some complications upstream that haven't been resolved yet that will effect this. Mainly that, we don't know yet is the vessel record source that the observer program is using is the same as the vessel record source that OLE is using, and there's potential for things like the vessel name to not match between one and the other, but both groups want to keep their version of the vessel name. So it's a bit TBD.